docs/affinityassistants.md

```diff
@@ -0,0 +1,116 @@
+weight: 405
+---
+-->
+
+# Affinity Assistants
+Affinity Assistant is a feature to coschedule `PipelineRun` `pods` to the same node
+based on [kubernetes pod affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity) so that it possible for the taskruns to execute parallel while sharing volume.
+Available Affinity Assistant Modes are **coschedule workspace**, **coschedule pipelineruns**,
+**isolate PipelineRun** and **disabled**. 
+
+* **coschedule workspace** - When a `PersistentVolumeClaim` is used as volume source for a `Workspace` in a `PipelineRun`,
+all `TaskRun` pods within the `PipelineRun` that share the `Workspace` will be scheduled to the same Node. 
+**Note:** Only one pvc-backed workspace can be mounted to each TaskRun in this mode.
+
+* **coschedule pipelineruns (WIP)** - All `TaskRun` pods within the `PipelineRun` will be scheduled to the same Node.
+
+* **isolate PipelineRun (WIP)** - All `TaskRun` pods within the `PipelineRun` will be scheduled to the same Node,
+and only one PipelineRun is allowed to run on a node at a time.
+
+* **disabled** - The Affinity Assistant is disabled. No pod coscheduling behavior.
+
+**Note:** **coschedule pipelineruns** and **isolate PipelineRun** modes are still under development and not yet functional.
+
+This means that Affinity Assistant is incompatible with e.g. other affinity rules
+ configured for the `TaskRun` pods. If the `PipelineRun` has a custom
+[PodTemplate](pipelineruns.md#specifying-a-pod-template) configured, the `NodeSelector` and `Tolerations` fields
+will also be set on the Affinity Assistant pod. The Affinity Assistant
+is deleted when the `PipelineRun` is completed. 
+
+Currently, the Affinity Assistant Modes can be configured by the `disable-affinity-assistant` and `coschedule` feature flags.
+In the future, the `disable-affinity-assistant` feature flag will be deprecated, and the Affinity Assistant Modes will be only determined by 
+the `coschedule` feature flag. 
+
+The following chart summarizes the Affinity Assistant Modes with different combinations of the `disable-affinity-assistant` and `coschedule` feature flags during migration (when both feature flags are present) and after the migration (when only the `coschedule` flag is present).
+ 
+**Note:** For users who previously accepted the default behavior (enabling affinity assistant) but now want one of the new features, you need to set `disable-affinity-assistant` to "true" and then turn on the new behavior by setting the `coschedule` flag. For users who previously disabled the affinity assistant but want one of the new features, just set the `coschedule` flag accordingly.
+
+**Note:** **DO NOT** change the default value of the `coschedule` feature flag since the feature is work in progress and not yet functional. Please **ONLY** toggle the `disabled-affinity-assistant` feature flag to switch between **"disabled"** and **"coschedule workspace"** modes.
```

Comment:
You may want to make this warning more prominent - moving this up to the top + ⚠️ / 🚨 emojis + bigger font / bold would help.

We should also try and classify all of these with their stability level to set expectations around breaking changes (by default users will assume stable).

Reply:
Good point! Will add the stability levels for different mode!

Reply:
Hi @wlynch. I have removed all the warnings and WIP in the documentation, and we will hold this PR until this feature is fully implemented (https://github.com/tektoncd/pipeline/pull/6927 is merged).

I have also added the stability levels for the feature, including adding it to the `alpha feature` list. PTAL 😄